web/src/js/components/footer.js

```diff
@@ -1,19 +1,18 @@
 import React from "react";
 import {SettingsState} from "./common.js";
 
-var Footer = React.createClass({
-    mixins: [SettingsState],
-    render: function () {
-        var mode = this.state.settings.mode;
-        var intercept = this.state.settings.intercept;
-        return (
-            <footer>
-                {mode && mode != "regular" ? <span className="label label-success">{mode} mode</span> : null}
-                &nbsp;
-                {intercept ? <span className="label label-success">Intercept: {intercept}</span> : null}
-            </footer>
-        );
-    }
-});
+Footer.propTypes = {
+    settings: React.PropTypes.object.isRequired,
+};
 
-export default Footer;
\ No newline at end of file
+export default function Footer({ settings }) {
+    const mode = settings.mode;
```

Comment:
Let's try to have a code style that is as uniform as possible. Do you prefer
```javascript
const mode = settings.mode;
const intercept = settings.intercept;
```
or 
```javascript
const {mode, intercept} = settings;
```
?